WMF Framework/Remote Notifications/RemoteNotificationsAPIController.swift

```diff
@@ -116,25 +117,64 @@ struct RemoteNotificationsAPIController {
 
     public func markAsRead(_ notifications: Set<RemoteNotification>, completion: @escaping (Error?) -> Void) {
         let maxNumberOfNotificationsPerRequest = 50
+        let notifications = Array(notifications)
 
-        guard notifications.count <= maxNumberOfNotificationsPerRequest else {
-            // TODO: Split requests? 50 is the limit.
-            assertionFailure()
-            return
-        }
-        
-        
-
-        request(Query.markAsRead(notifications: notifications), method: .post) { (result: MarkReadResult?, _, _, error) in
-            if let error = error {
-                completion(error)
+        if notifications.count > maxNumberOfNotificationsPerRequest {
+            let split = notifications.chunked(into: maxNumberOfNotificationsPerRequest)
+            let group = DispatchGroup()
+            var markAsReadError: Error? = nil
+            var failed = 0
+            for notifications in split {
```

Comment:
It looks like this could use `asyncMap`, or maybe a new variant `asyncCompactMap` to allow the error return value to be nil

Reply:
good point 👍